### PR TITLE
Bug 1219681 - fixed: successfully test unlocking with correct passcode and remaining locked with incorrect passcode

### PR DESCRIPTION
**Pull request by snowmantw**

### CODE REVIEW

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=a9d79ceab65979fe247beefa65fc3d700d673c85

File: apps/system/test/marionette/lib/lockscreen_passcode_unlock_actions.js [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+          return false;   // True: only when it rises to 100% fit the screen.
+        }
+      });
+    }).bind(this));
+  };
+
+  LockScreenPasscodeUnlockActions.prototype.pressKey = function(keyChar) {
+    var selector = '#lockscreen-passcode-pad a[data-key="' + keyChar + '"]';
+    // To wait it displayed.
+    this.client.helper.waitForElement(selector).click();
```
Comment (timdream) [inline apps/system/test/marionette/lib/lockscreen_passcode_unlock_actions.js]: `tap()`?

File: apps/system/test/marionette/lib/lockscreen_passcode_unlock_actions.js [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+        return !frame.displayed();
+      }).bind(this));
+      if (this.client.executeScript(function() {
+        return window.wrappedJSObject.Service.query('locked');
+      })) {
+        reject('Not unlocked before screen off');
+      } else {
+        resolve();
+      }
+    }).bind(this));
```
Comment (timdream) [inline apps/system/test/marionette/lib/lockscreen_passcode_unlock_actions.js]: Nice assertions here -- waiting for both UI change and internal state change.

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=486ad7e2d34cd33bc1f1a871cf533737e69665c2

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=8708ea99f690a1aaa26c0f05caf5e68e64fb7b65

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=e2c53fa7d2a3d874d7197d61e5b95cb9d702b473

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=06571ea23948b1a72dd34156012fe8cf072a8197

**Pull request merged**